VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -203,25 +207,25 @@ public void onComplete(FacebookDialog.PendingCall pendingCall, Bundle data) {
      * @param savedInstanceState
      */
     private void restore(Bundle savedInstanceState) {
-        try{
-            if (player == null) {
-                if (savedInstanceState != null
-                        && savedInstanceState.containsKey("player")) {
-                    player = (IPlayer) savedInstanceState.get("player");
-                } else {
-                    player = new Player();
-                }
-            }
+        if (player == null) {
             if (savedInstanceState != null
-                    && savedInstanceState.containsKey("isMessageDisplayed")){
-                if(savedInstanceState.getBoolean("isMessageDisplayed")){
-                    showVideoNotAvailable(VideoNotPlayMessageType.IS_VIDEO_MESSAGE_DISPLAYED);
-                }
+                    && savedInstanceState.containsKey("player")) {
+                player = (IPlayer) savedInstanceState.get("player");
+            } else {
+                player = new Player();
             }
-            reAttachPlayEventListener();
-        }catch(Exception e){
-            logger.error(e);
         }
+        if (savedInstanceState != null
+                && savedInstanceState.containsKey("isMessageDisplayed")) {
```

Comment:
As above, the `containsKey()` check is not necessary, as the `getBoolean()` method will return `false` if they key is not present.